homeassistant/components/homekit/__init__.py

```diff
@@ -0,0 +1,143 @@
+"""Support for Apple Homekit.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/homekit/
+"""
+import asyncio
+import logging
+
+import voluptuous as vol
+
+from pyhap.accessory_driver import AccessoryDriver
+
+from homeassistant.core import callback
+import homeassistant.helpers.config_validation as cv
+from homeassistant.const import (
+    CONF_NAME, CONF_IP_ADDRESS, CONF_PORT,
+    EVENT_HOMEASSISTANT_START, EVENT_HOMEASSISTANT_STOP)
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.entity_component import EntityComponent
+
+from .accessories import HomeBridge
+from .covers import Window
+from .sensors import TemperatureSensor
+
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = 'homekit'
+REQUIREMENTS = ['HAP-python==1.1.2']
+
+ATTR_CLASS = 'class'
+ATTR_NAME = 'name'
+
+CONF_PIN_CODE = 'pincode'
+CONF_TYPES = 'types'
+
+ALL_TYPES = {
```

Comment:
Instead of registering it like this, follow the example in Alexa and use our registry:

```python
# In __init__.py
from homeassistant.util.decorator import Registry

TYPES = Registry()
```

```python
# Then in cover.py
from . import TYPES

@TYPES.register('Cover')
class Window(HomeAccessory):
    …
```

You only have to import the cover file once in your setup and they will be added to your TYPES constant, to be used whenever you want.